### PR TITLE
Yang HW7 submission

### PR DESCRIPTION
**Pull request by YangJiang89**
HW 7
@MengyuanZoe @hanghu @jhp312

### CODE REVIEW

File: HW7/zipcode_map_Yang.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
+Folium interact with GeoJSON data
+Examples: overlay another GeoJSON zipcode map to the original map
+Author: Yang Jiang
+"""
+
+import pandas as pd
+import folium
+
```
Comment (MengyuanZoe) [inline HW7/zipcode_map_Yang.py:10]: Add one more blank line.

File: HW7/zipcode_map_Yang.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    return zipcode
+
+
+if __name__ == "__main__":
+    """
+    Example of using show_zipcode_map funciton
+    """
+
+    # Load King County house price data
+    house_data = pd.read_csv("../Data/kc_house_data.csv",
```
Comment (MengyuanZoe) [inline HW7/zipcode_map_Yang.py]: Rename the constant name. Avoid underscore. Use "HouseData".

File: HW7/zipcode_map_Yang.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    Example of using show_zipcode_map funciton
+    """
+
+    # Load King County house price data
+    house_data = pd.read_csv("../Data/kc_house_data.csv", 
+                             parse_dates=['date'])
+    house_data['zipcode'] = house_data['zipcode'].astype(str)
+
+    # Group data by zipcode and calculate the mean value in each zipcode
+    zipcode_data = pd.groupby(house_data, 'zipcode').mean()
```
Comment (MengyuanZoe) [inline HW7/zipcode_map_Yang.py:57]: Rename constant. Use "ZipcodeData".

File: HW7/zipcode_map_Yang.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    ----------
+    path : string
+        URL or File path to GeoJSON zipcode data
+
+    data : pandas dataframe
+        The other data set to interact with (house price or crime)
+
+    col : string
+        The colomn name in dataset to bound zipcode with
+    """
```
Comment (MengyuanZoe) [inline HW7/zipcode_map_Yang.py:32]: Also discuss the output/return value of this module in the comments.

File: HW7/zipcode_map_Yang.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    Parameters
+    ----------
+    path : string
+        URL or File path to GeoJSON zipcode data
+
+    data : pandas dataframe
+        The other data set to interact with (house price or crime)
+
+    col : string
+        The colomn name in dataset to bound zipcode with
```
Comment (puleo-joe) [inline HW7/zipcode_map_Yang.py]: Remove trailing whitespace in lines 25, 29, 30, 31, 43

File: HW7/zipcode_map_Yang.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    zipcode_data = pd.groupby(house_data, 'zipcode').mean()
+
+    # Add new fields in dataset, the count of house sold in each zipcode
+    zipcode_data['count'] = pd.groupby(house_data, 'zipcode').count()['id']
+    zipcode_data.reset_index(inplace=True)
+
+    # Path for GeoJSON zipcode data
+    zipcodepath = '../Data/zipcode_king_county.geojson'
+    show_zipcode_map(zipcodepath, zipcode_data, 'count')
+    show_zipcode_map(zipcodepath, zipcode_data, 'price')
```
Comment (puleo-joe) [inline HW7/zipcode_map_Yang.py:68]: Specify what the expected output for these maps are in a comment.

File: HW7/zipcode_map_Yang.py (hunk 5 above)
Comment (puleo-joe) [inline HW7/zipcode_map_Yang.py]: spelling error: Change "colomn" to "column"

File: HW7/zipcode_map_Yang.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        The other data set to interact with (house price or crime)
+
+    col : string
+        The colomn name in dataset to bound zipcode with
+    """
+    # Generate original map
+    zipcode = folium.Map(location=[data['lat'].mean(), 
+                                   data['long'].mean()], zoom_start=10)
+
+    # Add zipcode map layer to orignial map
```
Comment (puleo-joe) [inline HW7/zipcode_map_Yang.py:37]: spelling error:  Change "orignial" to "original"

File: HW7/zipcode_map_Yang.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+                       columns=['zipcode', col], 
+                       key_on='feature.properties.ZCTA5CE10', 
+                       fill_color='OrRd', fill_opacity=0.5, line_opacity=0.2)
+    zipcode.save('zipcode_' + col + '.html')
+    return zipcode
+
+
+if __name__ == "__main__":
+    """
+    Example of using show_zipcode_map funciton
```
Comment (puleo-joe) [inline HW7/zipcode_map_Yang.py]: Spelling error: Change "funciton" to "function"

File: HW7/zipcode_map_Yang.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    # Group data by zipcode and calculate the mean value in each zipcode
+    zipcode_data = pd.groupby(house_data, 'zipcode').mean()
+
+    # Add new fields in dataset, the count of house sold in each zipcode
+    zipcode_data['count'] = pd.groupby(house_data, 'zipcode').count()['id']
+    zipcode_data.reset_index(inplace=True)
+
+    # Path for GeoJSON zipcode data
+    zipcodepath = '../Data/zipcode_king_county.geojson'
+    show_zipcode_map(zipcodepath, zipcode_data, 'count')
```
Comment (MengyuanZoe) [inline HW7/zipcode_map_Yang.py:66]: Add more comments for two usages of the show_zipcode_map, e.g. what you are plotting.

File: HW7/zipcode_map_Yang.py (hunk 1 above)
Comment (YangJiang89) [reply HW7/zipcode_map_Yang.py:10]: Done

**Comment by YangJiang89** [comment]
Update my code based on the reviews:
fix the white blank, typos, add more comments
No change on the variable name

**Pull request merged**